### ISSUE TITLE
Unknown but valid SPDX 

### ISSUE THREAD

**Issue by BenBals** [haskell/cabal#6181]
**Describe the bug**
Cabal does not recognize the BlueOak-1.0.0 license even though that is a valid SPDX expression per https://spdx.org/licenses/BlueOak-1.0.0.html

**To Reproduce**
Use the following cabal file:
```cabal
cabal-version:       2.4

-- Initial package description 'turing.cabal' generated by 'cabal init'.
-- For further documentation, see http://haskell.org/cabal/users-guide/

-- The name of the package.
name:                turing

-- The package version.  See the Haskell package versioning policy (PVP)
-- for standards guiding when and how versions should be incremented.
-- https://pvp.haskell.org
-- PVP summary:      +-+------- breaking API changes
--                   | | +----- non-breaking API additions
--                   | | | +--- code changes with no API change
version:             0.1.0.0

-- A short (one-line) description of the package.
synopsis:            Encoding of turing machines

-- A longer description of the package.
-- description:

-- A URL where users can report bugs.
-- bug-reports:

-- The license under which the package is released.
license:             BlueOak-1.0.0

-- The file containing the license text.
license-file:        LICENSE.md

-- The package author(s).
author:              Ben Justus Bals

-- An email address to which users can send suggestions, bug reports, and
-- patches.
maintainer:          benbals@posteo.de

-- A copyright notice.
-- copyright:

category:            Language

-- Extra files to be distributed with the package, such as examples or a
-- README.
extra-source-files:  CHANGELOG.md


library
  -- Modules exported by the library.
  exposed-modules: Turing
                   Turing.Encode
                   Turing.Decode

  -- Modules included in this library but not exported.
  -- other-modules:

  -- LANGUAGE extensions used by modules in this package.
  -- other-extensions:

  -- Other library packages from which modules are imported.
  build-depends:       base ^>=4.12.0.0

  -- Directories containing source files.
  hs-source-dirs: lib

  -- Base language which the package is written in.
  default-language:    Haskell2010


executable turing
  -- .hs or .lhs file containing the Main module.
  main-is:             src/Main.hs

  -- Modules included in this executable, other than Main.
  -- other-modules:

  -- LANGUAGE extensions used by modules in this package.
  -- other-extensions:

  -- Other library packages from which modules are imported.
  build-depends:       base ^>=4.12.0.0

  -- Directories containing source files.
  hs-source-dirs: src

  -- Base language which the package is written in.
  default-language:    Haskell2010


```

**Expected behavior**
Cabal accepting the license.

**System informataion**
- cabal-install version 2.4.1.0
- Linux nixos-beb 4.19.58 #1-NixOS SMP Wed Jul 10 07:53:48 UTC 2019 x86_64 GNU/Linux

**Comment by phadej** [comment]
BlueOak is relatively new license and wasn’t in SPDX license list used by `cabal-version: 2.4`. It will be available with `cabal-version: 3.0`. https://github.com/haskell/cabal/pull/6171

**Comment by BenBals** [comment]
Thank you very much for the information.
